manifests/entry_script.pp

```diff
@@ -6,15 +6,25 @@
   String                            $base_dir,
   String                            $script_name,
   Integer[1]                        $index,
+  Boolean                           $use_python_venv = true,
+  Array[String]                     $parameters      = [],
   Enum['file','present','absent']   $ensure          = 'file',
 ) {
   # hook entry point
-  $bootstrap_lifecycle_hook_cmd = "#!/bin/bash\n${base_dir}/venv/bin/python ${base_dir}/${script_name}\n"
+  if $use_python_venv {
+    $cmd_init = "${base_dir}/venv/bin/python"
+  } else {
+    $cmd_init = undef
+  }
+
+  $cmd = join([$cmd_init, "${base_dir}/${script_name}", $parameters, "\n"], ' ')
+
+  $entry_script = join(['#!/bin/bash', $cmd], "\n")
```

Comment:
Minor remark: add `exec` in front of the command. There is no need to keep the shell around